🐛 Fixed redirects with special characters

Prathamesh010 opened the pull request:
refs/closes #15267
Custom redirect URLs with special characters showing 404 instead of redirection 
This was because the URLs were not being encoded before adding to the router.

- [x] There's a clear use-case for this code change, explained below
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test:all` and `yarn lint`)

The issue is solved by encoding the URL before its added to the router.

```
301:
    /joloonii-surgaltuud/b-angilal/: /joloonii-angilal/#в-ангилал
    /joloonii-surgaltuud/а-ангилал/: /joloonii-angilal/#а-ангилал
```
Here, the first one resolves perfectly as it does not have any special characters but the second one fails to redirect.

I tried adding the encoded version of `/joloonii-surgaltuud/а-ангилал/` ie.
`/joloonii-surgaltuud/%D0%B0-%D0%B0%D0%BD%D0%B3%D0%B8%D0%BB%D0%B0%D0%BB` into the yaml and it resolved perfectly. 

So the issue was in encoding as `/joloonii-surgaltuud/а-ангилал/` never existed on the router but the encoded one did.

ErisDS reviewed (changes requested):
Hi @Prathamesh010, thanks for your PR!

I believe that this should be handled inside the redirect manager itself? 

I think that will also make it easier to add unit tests that demonstrate the change is correct, which we would need in order to safely merge this please.

Prathamesh010 committed:
🐛 Encoded custom-redirect uri before adding it to router
refs/closes #
custom redirect urls with special characters showing 404 instead of redirection
this was because of the urls not being encoded before adding to the router

Prathamesh010 committed:
Added tests

ErisDS commented:
Hi @Prathamesh010 thanks so much for taking the time to create and update this PR. The updated solution looks great 👌 

This has now been merged 🎉 and will appear in the next release of Ghost - usually Fridays.

I'm not sure if you found this through hacktoberfest, but I've added the accepted label to this PR to make sure it counts.

We have plenty of open issues and ongoing projects for hacktoberfest and beyond, including a project around [i18n](https://github.com/TryGhost/Ghost/issues/15502) and there was also a recent discussion about adding support for special characters in more [key parts of Ghost](https://github.com/TryGhost/Ghost/issues/14662) that I can reopen if it's interesting to you as we're looking for experts to take on this work and help us make sure it doesn't break anything 🙂
